homeassistant/components/mystrom/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow for myStrom integration."""
+from __future__ import annotations
+
+import ipaddress
+import logging
+import re
+from typing import Any
+
+import pymystrom
+from pymystrom.exceptions import MyStromConnectionError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+
+def host_valid(host):
+    """Return True if hostname or IP address is valid."""
+    try:
+        if ipaddress.ip_address(host).version == (4 or 6):
+            return True
+    except ValueError:
+        disallowed = re.compile(r"[^a-zA-Z\d\-]")
+        return all(x and not disallowed.search(x) for x in host.split("."))
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    if not host_valid(data[CONF_HOST]):
+        raise CannotConnect()
+    try:
+        info = await pymystrom.get_device_info(data[CONF_HOST])
+    except MyStromConnectionError as error:
+        raise CannotConnect() from error
+
+    return {"mac": info.get("mac")}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for myStrom."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Handle import from config."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(info["mac"])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(title=info["mac"], data=user_input)
```

Comment:
Unfortunatelly, there is no other information from the device api available except the mac address and the ip address. In a previous very, I had implemented a configurable name (which is also the case before this PR: The name is set in the configuration.yaml).
Would it make sense to implement the name config option again? From my perspective, this would be the most user friendly option given that such information is not available in the api. @epenet What do you think?

Reply:
I guess you can ask for a name, and store that in the entry title.
You shouldn't store it in the config data.
You can then use the entry title for naming devices and entities.

Reply:
I gave it a try and implemented a title. Would this work?